Lexis+.js

```diff
@@ -0,0 +1,221 @@
+				else if (potentialReporter.textContent.match(/pub\./i)
+				|| potentialReporter.textContent.match(/p\.l\./i)) {
+					publicLawNo = potentialReporter.textContent;
+				}
+			}
+
+			var otherReporters = doc.querySelectorAll('span.SS_NonPaginatedRptr');
+
+			for (var i = 0; i < otherReporters.length; i++) {
+				var nextReporter = otherReporters[i].textContent;
+				if (nextReporter.match(/stat\./i)) statutesAtLarge = nextReporter;
+				else if (nextReporter.match(/pub\./i)
+				|| nextReporter.match(/p\.l\./i)) {
+					publicLawNo = nextReporter;
+				}
+			}
+
+			// Turn publicLawNo into the public law fields
+			if (publicLawNo.match(/\d+-\d+/)) { // Ex. P.L. 115-164
+				var numPos = publicLawNo.search(/\d+-\d+/);
+				newStatute.publicLawNumber = publicLawNo.substring(numPos, publicLawNo.substring(numPos + 1).indexOf(' ')); // Gets 115-164
+
+				newStatute.session = newStatute.publicLawNumber.substring(0, newStatute.publicLawNumber.indexOf('-'));
+			}
+			else { // Ex. 115 P.L. 164 or 115 Pub. L. No. 164
+				newStatute.session = publicLawNo.substring(0, publicLawNo.indexOf(' '));
+				newStatute.publicLawNumber = newStatute.session + '-' + publicLawNo.substring(publicLawNo.lastIndexOf(' ') + 1);
+			}
+
+			// Turn statutesAtLarge into the code#/code/section fields
+			// TODO in styles, check for "Stat." as the code, and if so, don't append a section symbol
+			newStatute.codeNumber = statutesAtLarge.substring(0, statutesAtLarge.indexOf(' '));
+			newStatute.code = "Stat.";
+			newStatute.section = statutesAtLarge.substring(statutesAtLarge.lastIndexOf(' ') + 1);
+		}
+		else { // Codified statute
+			if (title.match(/^\d+/)) { // Starts with digit, organized by title, ex. 47 U.S.C.S. § 230
+				// Sadly, named groups aren't working
+				let groups = title.match(/^(\d+)\s([a-zA-Z0-9. ]+) § ([0-9.()a-zA-Z]+)/);
+				newStatute.codeNumber = groups[1];
+				newStatute.code = groups[2];
+				newStatute.section = groups[3];
+			}
+			else { // Starts with letter, organized by code, ex. Tex. Bus. & Com. Code § 26.01
+				let groups = title.match(/^([a-zA-Z. ]+) § ([0-9.()a-zA-Z]+)/);
+				newStatute.code = groups[1];
+				newStatute.section = groups[2];
+			}
+
+			// No way to tell which will be present
+			var pL = info.match(/p\.l\. (\d+-\d+)/i);
+			var pubLaw = info.match(/pub\. law (\d+-\d+)/i);
+			var pubLawNo = info.match(/pub\. law no\. (\d+-\d+)/i);
+			var publicLaw = info.match(/public law (\d+-\d+)/i);
+			publicLawNo = info.match(/public law no\. (\d+-\d+)/i);
+			var publicLawNumber = info.match(/public law number (\d+-\d+)/i);
+			if (pL) newStatute.publicLawNumber = pL[1];
+			if (pubLaw) newStatute.publicLawNumber = pubLaw[1];
+			if (pubLawNo) newStatute.publicLawNumber = pubLawNo[1];
+			if (publicLaw) newStatute.publicLawNumber = publicLaw[1];
+			if (publicLawNo) newStatute.publicLawNumber = publicLawNo[1];
+			if (publicLawNumber) newStatute.publicLawNumber = publicLawNumber[1];
```

Comment:
This can be a lot simpler. It could be combined into a single regex:

```javascript
var publicLawNumber = info.match(/(?:p\.l\.|pub(?:\.|lic) law(?: no\.| number)) (\d+-\d+)/i);
```

with the possible options in a comment above.

(Or even as multiple, `match` returns null if no match, so you could chain them with `||` as assignments to a single variable.)